include/miral/miral/window_manager_tools.h

```diff
@@ -39,229 +39,273 @@ struct ApplicationInfo;
+/// the policy will receive an instance of #miral::WindowManagerTools as the first parameter.
+/// Compositor authors are encouraged to use this class instance to handle and request changes from
+/// the underlying Mir server.
 class WindowManagerTools
 {
 public:
+    /// Constructs an instance of window manager tools from an implementation.
+    ///
+    /// Compositor authors are not expected to construct this object themselves.
+    /// Instead, they should use #miral::add_window_manager_policy to create
+    /// an instance of this class for them. For this reason,
+    /// #miral::WindowManagerToolsImplementation is purposefully opaque.
     explicit WindowManagerTools(WindowManagerToolsImplementation* tools);
     WindowManagerTools(WindowManagerTools const&);
     WindowManagerTools& operator=(WindowManagerTools const&);
     ~WindowManagerTools();
 
-/** @name Query & Update Model
- *  These functions assume that the BasicWindowManager data structures can be accessed freely.
- *  I.e. they should only be used by a thread that has called the WindowManagementPolicy methods
- *  (where any necessary locks are held) or via a invoke_under_lock() callback.
- *  @{ */
-
-    /** count the applications
-     *
-     * @return number of applications
-     */
+    /// Returns the number of applications.
+    ///
+    /// @returns the number of applications
     auto count_applications() const -> unsigned int;
 
-    /** execute functor for each application
-     *
-     * @param functor the functor
-     */
+    /// Execute a functor for each application.
+    ///
+    /// \param functor called for each application
     void for_each_application(std::function<void(ApplicationInfo& info)> const& functor);
 
     /** find an application meeting the predicate
      *
      * @param predicate the predicate
      * @return          the application
      */
+    /// Find an application that meets the predicate.
+    ///
+    /// \param predicate the predicate
+    /// \returns an application, or the default-constructed application if none is found
     auto find_application(std::function<bool(ApplicationInfo const& info)> const& predicate)
     -> Application;
 
-    /** retrieve metadata for an application
-     *
-     * @param session   the application session
-     * @return          the metadata
-     */
+    /// Retrieve information about a \p session.
+    ///
+    /// \param session the application
+    /// \returns info about the application, or the default-constructed application info if the \p session is invalid
+    /// \sa miral::ApplicationInfo - info about the application
     auto info_for(std::weak_ptr<mir::scene::Session> const& session) const -> ApplicationInfo&;
 
-    /** retrieve metadata for a window
-     *
-     * @param surface   the window surface
-     * @return          the metadata
-     */
+    /// Retrieve info about a window from the \p surface.
```

Comment:
Elsewhere "metadata" has been rephrased to "information" is there a reason for this inconsistency?